### PR TITLE
feature: add twice interacition when sealer delete

### PR DESCRIPTION
**Pull request by tea403**
when use sealer delete . Its a dangerous comand. it should be twice interacition with user.

### CODE REVIEW

File: sealer/cmd/delete.go [hunk 1]
```diff
@@ -22,13 +22,30 @@ import (
...
+
 // deleteCmd represents the delete command
 var deleteCmd = &cobra.Command{
 	Use:     "delete",
 	Short:   "delete a cluster",
 	Long:    `if provider is BARESERVER will delete kubernetes nodes, or if provider is ALI_CLOUD, will delete all the infra resources`,
 	Example: `sealer delete -f /root/.sealer/mycluster/Clusterfile`,
 	Run: func(cmd *cobra.Command, args []string) {
+		var input string
+		for {
```
Comment (fanux) [inline sealer/cmd/delete.go:38]: 1. "y" or "yes"
2. We also can provide an --force flag to delete cluster by force.

Comment (oldthreefeng) [reply sealer/cmd/delete.go:38]: LGTM .   @tea403 

a regex using may be ok.  

```
package main

import (
	"fmt"
	"regexp"
)

func main() {

	var yesRx = regexp.MustCompile("^(?i:y(?:es)?)$")

	var input string
	for {
		fmt.Scanln(&input)
		if yesRx.MatchString(input) {
			fmt.Println("regex match yes")
			break
		} else {
			fmt.Println("regex match not")
		}
	}
}
```

Comment (tea403) [reply sealer/cmd/delete.go]: i already recommit the pr

File: sealer/cmd/delete.go [hunk 2]
```diff
@@ -27,8 +29,29 @@ var deleteCmd = &cobra.Command{
...
+			var input string
+			for {
+				fmt.Printf("Are you sure you want to delete the cluster? Yes [y/yes], No [n/no] : ")
+				fmt.Scanln(&input)
+				if yesRx.MatchString(input) {
+					break
+				}
+				if noRx.MatchString(input) {
+					fmt.Println("You have canceled to delete the cluster!")
+					os.Exit(1)
```
Comment (allencloud) [inline sealer/cmd/delete.go:51]: why to exit with a non-zero exit code?

Comment (tea403) [reply sealer/cmd/delete.go]: it changed

**Review by fanux — APPROVED** [review]

File: sealer/cmd/delete.go [hunk 3]
```diff
@@ -27,8 +29,29 @@ var deleteCmd = &cobra.Command{
...
+		if err != nil {
+			logger.Error(err)
+			os.Exit(1)
+		}
+		if !force {
+			var yesRx = regexp.MustCompile("^(?:y(?:es)?)$")
+			var noRx = regexp.MustCompile("^(?:n(?:o)?)$")
+			var input string
+			for {
+				fmt.Printf("Are you sure you want to delete the cluster? Yes [y/yes], No [n/no] : ")
```
Comment (fanux) [inline sealer/cmd/delete.go:44]: Are you sure you want to delete the cluster? -> Are you sure to delete the cluster?

Comment (tea403) [reply sealer/cmd/delete.go]: i get it

Comment (oldthreefeng) [reply sealer/cmd/delete.go]: LGTM. if rebase the commit.

**Pull request merged**